programs/util.c

```diff
@@ -130,13 +130,26 @@ int UTIL_getFileStat(const char* infilename, stat_t *statbuf)
     return 1;
 }
 
-/* like chmod, but avoid changing permission of /dev/null */
-int UTIL_chmod(char const* filename, mode_t permissions)
+int UTIL_chmod(char const* filename, mode_t permissions, int onlyRegularFiles)
 {
-    if (!strcmp(filename, "/dev/null")) return 0;   /* pretend success, but don't change anything */
+    if (onlyRegularFiles && !UTIL_isRegularFile(filename)) {
+        return 0;
+    }
     return chmod(filename, permissions);
 }
 
+int UTIL_chown(char const* filename, uid_t owner, gid_t group, int onlyRegularFiles) {
+    if (onlyRegularFiles && !UTIL_isRegularFile(filename)) {
```

Comment:
If the only known usage of `UTIL_chown()` sets `onlyRegularFiles == 0`, then this test could be replaced by an `assert()`.